src/app/utils.ts

```diff
@@ -0,0 +1,108 @@
+import path from 'path'
+import fs from 'fs'
+import os from 'os'
+
+export const UPLOAD_DIR = path.resolve(os.tmpdir(), 'public/uploads')
+export const metadataPath = path.resolve(UPLOAD_DIR, `metadata.json`)
+
+export const createUploadDirIfNotExists = async () => {
+    try {
+        if (!fs.existsSync(UPLOAD_DIR)) {
+            fs.mkdirSync(UPLOAD_DIR)
+        }
+    } catch (err) {
+        console.error('Error creating directory:', err)
+    }
+}
+
+export const loadMetadata = async () => {
+    let metadata: any = []
+    if (fs.existsSync(metadataPath)) {
+        metadata = JSON.parse(fs.readFileSync(metadataPath, 'utf8'))
+    }
+    return metadata
+}
+
+export const getFileExtension = (filename: string) => {
+    const parts = filename.split('.')
+    return parts.length > 1 ? parts.pop()?.toUpperCase() : 'N/A'
+}
+
+// TODO Override runID if already exists
+export const saveMetadata = async (file: Blob, name: string, runId: string) => {
+    const uploaded_metadata = {
+        runId: runId,
+        name: name,
+        size: file.size,
+        mimeType: file.type,
+        time: new Date().toISOString(),
+        fileType: getFileExtension(name),
+    }
+    let metadata = await loadMetadata()
+    metadata.push(uploaded_metadata)
+    try {
+        fs.writeFileSync(metadataPath, JSON.stringify(metadata))
+    } catch (err) {
+        console.error('Error writing metadata:', err)
+    }
+    return metadata
+}
+
+export const saveFile = async (file: Blob, name: string, runId: string) => {
+    await createUploadDirIfNotExists()
+    await saveMetadata(file, name, runId)
+    const buffer = Buffer.from(await file.arrayBuffer())
+    try {
+        fs.writeFileSync(path.resolve(UPLOAD_DIR, runId), buffer)
+    } catch (err) {
+        console.error('Error writing file:', err)
+    }
+}
+
+export const parseCSV = (data: string) => {
+    let lines = data.split('\n')
```

Comment:
Might be better to include a CSV library. For example, I'm not sure this supports commas inside strings.

Reply:
I will update with a CSV library. We ran into an issue with different delimiters yesterday.